jumpscale/clients/taiga/taiga.py

```diff
@@ -217,3 +224,302 @@ def _get_project_status(project_statuses, status):
+            all_projects_template (dic): dict of the data required from project object
+
+        """
+        all_projects_template = list()
+        for project in projects:
+            single_project_template = dict()
+            single_project_template["name"] = project.name
+            single_project_template["created_date"] = project.created_date
+            single_project_template["owner"] = project.owner.get("full_name_display")
+            single_project_template["issues"] = project.list_issues()
+            all_projects_template.append(single_project_template)
+        return all_projects_template
+
+    def __render_projects(self, text="", projects=None, with_details=False):
+        """Do all the render for project details that will written in markdown files.
+
+        Args:
+            text(str): the text will be used to append data to it.
+            projects (list): list of projects will be used in render
+            with_details (bool): flag used to get the issue details in case of true.
+
+        Returns:
+            text (str): the rendered text that will be writted to markdown file.
+        """
+        for project in projects:
+            text += f"##### {project.get('name')} \n"
+            issues = project.get("issues")
+
+            if with_details:
+                all_issues_templates = self.__get_issues_required_data(issues, with_description=True)
+                text = self.__render_issues_with_details(text=text, issues=all_issues_templates, with_description=True)
+            else:
+                text = self.__render_issues_with_subjects(text=text, issues=issues)
+
+        return text
+
+    def map_render_issues_per_project(self, with_details):
+        """Map the desired data from issues per project and render them to be used in markdown files.
+
+        Args:
+            with_details (bool): flag used to get the issue details in case of true.
+        """
+        projects = self.list_all_projects()
+        all_projects_template = self.__get_project_required_data(projects)
+        grouped_projects = self.__group_data(data=all_projects_template, grouping_attribute="owner")
+
+        for owner, projects_per_owner in grouped_projects.items():
+            self.text += f"## Owner: {owner} \n"
+            # Sort the issues per group
+            sorted_list = sorted(projects_per_owner, key=lambda x: x.get("created_date"), reverse=True)
+            self.text = self.__render_projects(text=self.text, projects=sorted_list, with_details=with_details)
+
+    def export_issues_per_project(self, path="/tmp/issues_per_project.md", with_details=False):
+        """Export issues per project in a markdown file
+        
+        Args:
+            path (str): The path of exported markdown file.
+            
+        """
+        self.text = """
+### Issues  Per Project \n
+"""
+
+        greenlet = gevent.spawn(self.map_render_issues_per_project, with_details)
+        gevent.joinall([greenlet])
```

Comment:
Actually, this is just like doing it outside a greenlet, as I see, we gain nothing by spawning a single greenlet and waiting for it.

I think the correct way is to split rendering of issues a cross a pool and wait for all of them instead.